rocAL/rocAL/source/loaders/image/image_read_and_decode.cpp

```diff
@@ -131,6 +134,26 @@ ImageReadAndDecode::set_batch_random_bbox_crop_coords(std::vector<std::vector<fl
     _crop_coords_batch = crop_coords;
 }
 
+void ImageReadAndDecode::feed_external_input(std::vector<std::string> input_images_names, std::vector<int> labels, std::vector<unsigned char *> input_buffer,
+                             std::vector<unsigned> roi_width, std::vector<unsigned> roi_height,
+                             unsigned int max_width, unsigned int max_height, int channels,  FileMode mode, bool eos)
+{
+    std::vector<size_t> image_size;
+    image_size.reserve(roi_height.size());
+    for(unsigned int i = 0; i < roi_height.size(); i++) {
+        if (mode == 2)
+            image_size[i] = (roi_width[i] * roi_height[i] * channels);
+        else if (mode == 1)
+            image_size[i] = roi_height[i];
```

Comment:
what happens if mode == 0 here?
image_size buffer need not be filled?

Reply:
image size is not required since we are passing only filenames

Reply:
Done